### ISSUE TITLE
Filter jetpack_search_should_handle_query does not work

### ISSUE THREAD

**Issue by htdat** [Automattic/jetpack#10567]
#### tl;dr: with the current code, `jetpack_search_should_handle_query` does nothing. 

This is the place where [jetpack_search_should_handle_query](https://developer.jetpack.com/hooks/jetpack_search_should_handle_query/ is applied

https://github.com/Automattic/jetpack/blob/bb884c04b32a57e3053f14c56e2f4e0138aef185/modules/search/class.jetpack-search.php#L455-L459

```
		if ( ! apply_filters( 'jetpack_search_should_handle_query', ( $query->is_main_query() && $query->is_search() ), $query ) ) {
			return $posts;
		}
		$this->do_search( $query );
```

Right after the filter, we call the do_search() method and in the first lines of this method we just allow the main search query to run: 

https://github.com/Automattic/jetpack/blob/bb884c04b32a57e3053f14c56e2f4e0138aef185/modules/search/class.jetpack-search.php#L502-L506

```
	public function do_search( WP_Query $query ) {
		if ( ! $query->is_main_query() || ! $query->is_search() ) {
			return;
		}
```

I think we need to remove the conditional in do_search().

**Comment by Viper007Bond** [comment]
It's not supposed to do anything by default but if you hook into it, it looks like it'd work fine?

```php
add_filter( 'jetpack_search_should_handle_query', '__return_false' );
```

**Comment by htdat** [comment]
Ah, I mean the condition check in do_search(). 

**Assume my target: let all WP_Query queries handled by ES.** 

I am you're using the following filter:

> add_filter( 'jetpack_search_should_handle_query', '__return_true' );

 All queries can go and execute $this->do_search(): 

https://github.com/Automattic/jetpack/blob/bb884c04b32a57e3053f14c56e2f4e0138aef185/modules/search/class.jetpack-search.php#L459

However, the `do_search()` method itself just lets the main search queries go through. In other words, my target can not reached. 

**What I suggest here is that we remove the conditinal check in `do_search()`.** 

Does that make sense now? (Sorry for the confusion if I may not use the correct title here).
